client/state/sites/selectors.js

```diff
@@ -20,17 +21,41 @@ import {
 import createSelector from 'lib/create-selector';
 import { fromApi as seoTitleFromApi } from 'components/seo/meta-title-editor/mappings';
 import versionCompare from 'lib/version-compare';
+import getComputedAttributes from 'lib/site/computed-attributes';
 
 /**
- * Returns a site object by its ID.
+ * Returns a normalized site object by its ID.
  *
  * @param  {Object}  state  Global state tree
  * @param  {Number}  siteId Site ID
  * @return {?Object}        Site object
  */
-export function getSite( state, siteId ) {
+export const getSite = createSelector(
+	( state, siteId ) => {
+		const site = state.sites.items[ siteId ];
+
+		if ( site ) {
+			Object.assign( site, getComputedAttributes( site ) );
+			site.hasConflict = isSiteConflicting( state, siteId );
+			site.slug = getSiteSlug( state, siteId );
+			site.domain = getSiteDomain( state, siteId );
+		}
+
+		return site || null;
```

Comment:
Let's just return early with null if the site is not set so we don't have the extra level of indentation in the condition above.